src/mongocxx/exception/base.hpp

```diff
@@ -16,13 +16,41 @@
 
 #include <mongocxx/config/prelude.hpp>
 
+#include <cstdint>
 #include <exception>
+#include <tuple>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+
+#include <mongocxx/stdx.hpp>
 
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace exception {
 
-class MONGOCXX_API base : public std::exception {};
+using error_and_code_t = std::tuple<std::string, std::int32_t>;
```

Comment:
since it has a meaning for all exceptions, I'd like to leave it in the exception namespace.